drivers/dac/dac_dac161s997.c

```diff
@@ -0,0 +1,311 @@
+	}
+}
+
+static int dac161s997_init(const struct device *dev)
+{
+	const struct dac161s997_config *config = dev->config;
+	struct dac161s997_data *data = dev->data;
+	union dac161s997_status status;
+	int ret;
+
+	data->dev = dev;
+
+	if (!spi_is_ready_dt(&config->bus)) {
+		LOG_ERR("SPI bus %s not ready", config->bus.bus->name);
+		return -ENODEV;
+	}
+
+	k_sem_init(&data->lock, 0, 1);
+
+	ret = dac161s997_write_reg(dev, DAC161S997_REG_RESET, 0xc33c);
+	if (ret != 0) {
+		return ret;
+	}
+	ret = dac161s997_write_reg(dev, DAC161S997_REG_NOP, 0);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* Read status to clear any sticky error caused during boot or reboot */
+	ret = dac161s997_read_status(dev, &status);
+	if (ret != 0) {
+		return ret;
+	}
+
+	/* Check that DAC_RES bits are all set */
+	if (status.dac_resolution != 0x7) {
+		LOG_ERR("Unexpected DAC resolution value: 0x%02x", status.dac_resolution);
+		return ret;
+	}
+
+	if (config->gpio_errb.port != NULL) {
+		if (!gpio_is_ready_dt(&config->gpio_errb)) {
+			LOG_ERR("ERRB GPIO is not ready");
+			return -ENODEV;
+		}
+
+		k_work_init(&data->gpio_errb_work, dac161s997_gpio_errb_work_handler);
+
+		gpio_init_callback(&data->gpio_errb_cb, dac161s997_gpio_errb_cb,
+				   BIT(config->gpio_errb.pin));
+
+		ret = gpio_pin_configure_dt(&config->gpio_errb, GPIO_INPUT);
+		if (ret != 0) {
+			LOG_ERR("Configure ERRB GPIO failed: %d", ret);
+			return ret;
+		}
+
+		ret = gpio_pin_interrupt_configure_dt(&config->gpio_errb, GPIO_INT_EDGE_TO_ACTIVE);
+		if (ret) {
+			LOG_ERR("Configure ERRB interrupt failed: %d", ret);
+			return ret;
+		}
+
+		ret = gpio_add_callback_dt(&config->gpio_errb, &data->gpio_errb_cb);
+		if (ret != 0) {
+			LOG_ERR("Configure ERRB callback failed: %d", ret);
+			return ret;
+		}
+	}
+
+	k_sem_give(&data->lock);
+
+	return 0;
+}
+
+static DEVICE_API(dac, dac161s997_driver_api) = {
+	.channel_setup = dac161s997_channel_setup,
+	.write_value = dac161s997_write_value,
+};
+
+#define DAC_DAC161S997_INIT(n)                                                                     \
+	static const struct dac161s997_config dac161s997_config_##n = {                            \
+		.bus = SPI_DT_SPEC_INST_GET(n, SPI_TRANSFER_MSB | SPI_WORD_SET(8), 0),             \
+		.gpio_errb = GPIO_DT_SPEC_INST_GET_OR(n, errb_gpios, {0}),                         \
+	};                                                                                         \
+                                                                                                   \
+	static struct dac161s997_data dac161s997_data_##n;                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(n, dac161s997_init, NULL, &dac161s997_data_##n,                      \
+			      &dac161s997_config_##n, POST_KERNEL,                                 \
+			      CONFIG_DAC_DAC161S997_INIT_PRIORITY, &dac161s997_driver_api);
```

Comment:
```suggestion
			      CONFIG_DAC_INIT_PRIORITY, &dac161s997_driver_api);
```

Reply:
Will a device that depends on another always be initialized after even if it has equal priority?

Reply:
Still yes ;-)

You can check the initialisation order after building with `west build -t initlevels`.